ctapipe/reco/FitGammaHillas.py

```diff
@@ -0,0 +1,308 @@
+from itertools import combinations,permutations
+
+import numpy as np
+
+from scipy.optimize import minimize
+
+from astropy import units as u
+u.dimless = u.dimensionless_unscaled
+
+from ctapipe.utils import linalg
+
+from ctapipe.reco.reco_algorithms import RecoShowerGeomAlgorithm
+from ctapipe.io.containers import RecoShowerGeom
+
+
+
+__all__ = ["FitGammaHillas"]
+
+
+
+
+def guessPixDirection(pix_x, pix_y, tel_phi, tel_theta, tel_foclen, camera_rotation=-100.893 * u.degree):
+    # TODO replace with proper implementation
+    # beta is the pixel's angular distance to the centre according to beta / tel_view = r / maxR
+    # alpha is the polar angle between the y-axis and the pixel
+    # to find the direction the pixel is looking at, 
+    #  - the pixel direction is set to the telescope direction
+    #  - offset by beta towards up
+    #  - rotated around the telescope direction by the angle alpha
+    
+    pix_alpha = np.arctan2(pix_x, pix_y)
+    pix_beta  = (pix_x**2 + pix_y**2)**.5
+
+    pix_beta  = pix_beta / tel_foclen * u.rad 
+    
+    tel_dir = linalg.set_phi_theta(tel_phi,tel_theta)
+
+    pix_dirs = []
+    
+    
+    for a, b in zip(pix_alpha,pix_beta):
+        pix_dir = linalg.set_phi_theta( tel_phi, tel_theta + b )
+        
+        pix_dir = linalg.rotate_around_axis(pix_dir, tel_dir, (a-camera_rotation))
+        pix_dirs.append(pix_dir*u.dimless)
+        
+    return pix_dirs
+
+
+class FitGammaHillas(RecoShowerGeomAlgorithm):
+    
+    def __init__(self):
+        self.tel_geom = {}
+        self.circles = {}
```

Comment:
this doesn't match the `Component` `__init__`.  If you want  to define an __init__ you must call `super().__init__(...)`  to make sure the superclass is initialized correctly.  So far you aren't really using the `Component` functionality however, so you could just remove it as the base class, but if you ever want any user-configurable parameters, it is useful to have. If there are no parameters, or the paraemters can be added by wrapping the whole thing in a `Container`, then it's fine to leave the base class  of `RecoShowerGeomAlgorithm` empty.